### DISCUSSION TITLE
Spherical Custom Landscape + Polygonal Landscape

### DISCUSSION THREAD

**Discussion by rac19** [Stellarium/stellarium#1887] [Q&A] [answered]
I have successfully implemented a spherical landscape using a panorama of my backyard and I am very impressed with how useful (and accurate) it for avoiding obstructions in my FOV, It also helps considerable with visual orientation,

I have attempted to superimpose a polygonal horizon onto it, which I believe is possible, but so far without success. The spherical landscape was very useful in creating the polygonal_horizon_list file which displays perfectly as a standalone landscape but so far I have not been able to combine `them.`

Also, I am not sure what to expect if I succeed. I would like to see the original panoramic image with the polygon horizon line closely following the panorama horizon except when I have deviated to should the outline of some tall trees. The would require the ground_color to be transparent and I am not sure if that is possible.

Here is the configuration so far. Any advice that can be offered would be much prevaricated.

[landscape]
name = Backyard
description = Backyard
type = spherical
maptex = Backyard.png
maptex_top = 46
maptex_bottom = -4
altitude = 90
angle_rotatez=90
polygonal_horizon_list = BackyardHorizon.txt
polygonal_horizon_list_mode = azDeg_altDeg
polygonal_angle_rotatez = 0
horizon_line_color = 0.15,0.45,0.45

``

**Comment by gzotti** [comment]
There is no altitude entry in the landscape description. Maybe for your [location] (amsl)?
The combination should appear as photo panorama with polygonal line. 
You could add a ground_color to fill the area below -4°.
If something does not work, the logfile may contain hints.

**Reply by rac19** [reply]
Thanks, the location and elevation were present and I removed them to see if they were causing a problem. I can put them back in. Previously, all three parameters were in the [landscape] section, as per an example that I had seen. I will try a using a [location] section.

At the moment, with the ini file as shown, the spherical landscape works but not the polygonal landscape.

I will check the log to see if there is any useful information.

**Comment by rac19** [comment, marked as answer]
I have found the solution. There is apparently a bug that comes into play when **polygonal_angle_rotatez = 0**. Changing it to **polygonal_angle_rotatez = 0.0001** fixed it

**Reply by gzotti** [reply]
Ah yes, that one. Good that you found it.

**Comment by rac19** [comment]
Just for the record, this is an example of what it looks like. There is a small bug with the polygon which seems to have taken a shortcut to close the loop. I imagine that adding a few more points at 0 degrees around the horizon might fix it, but it looks OK to me as it is.
![Panorama3](https://user-images.githubusercontent.com/20808962/133360939-009c0ca9-b06d-485f-9bda-9a61556eb877.jpeg)
